index.html

```diff
@@ -185,18 +172,16 @@ <h4>Association and activation</h4>
                 </p>
             </div>
             <p>
-                A [=top-level traversable=] should have at most one <dfn>active EditContext</dfn>, which is determined
-                by running the steps to [=determine the active EditContext=] for that [=top-level traversable=].
+                A {{Document}} has an <dfn>active EditContext</dfn>, which may be null.
             </p>
             <p>
-                When an {{EditContext}} that was previously the [=active EditContext=] stops being
-                the [=active EditContext=], run the steps to [=deactivate an EditContext=]
-                on that {{EditContext}}.
+                When a {{Document}} receives text input from the [=Text Input Service=] and that
```

Comment:
Do we need to define what "text input" is? It could be IME or input from keyboard or handwriting composition or dictation etc. On Windows, TSF handles all the inputs that involves composition (handwriting, dictation, IMEs etc), but I'm not sure if that is true for all platforms. On Android, I believe regular keyboard typing comes as a composition event, so we may need to find a way to define these terms without being too focused on one platform.

Reply:
This does need to be defined more precisely, and I think the entry point for `Update the EditContet` should actually be over in the [input events spec ](https://w3c.github.io/input-events/#input-event-order-during-composition). There will probably be at least a couple different calls from there, one for composition and one for "normal" typing.

This PR is already too big so I'd like to postpone it for now, but this is definitely part of my planned work.